tests/unit/prediction/ocr/test_image_flag.py

```diff
@@ -0,0 +1,153 @@
+import pytest
+import requests
+from openfoodfacts.ocr import OCRResult
+
+from robotoff.prediction.ocr.image_flag import flag_image
+from robotoff.types import PredictionType
+
+
+@pytest.fixture
+def face_detection_ocr_result():
+    test_data_url = "https://raw.githubusercontent.com/openfoodfacts/test-data/main/robotoff/tests/unit/ocr/face_detection.json"
+    response = requests.get(test_data_url)
+    response.raise_for_status()
+    return OCRResult.from_json(response.json())
+
+
+def test_flag_image_with_face_annotation(face_detection_ocr_result):
+    predictions = flag_image(face_detection_ocr_result)
+
+    assert predictions
+
+    face_predictions = [
+        pred
+        for pred in predictions
+        if pred.type == PredictionType.image_flag
+        and pred.data.get("type") == "face_annotation"
+    ]
+
+    assert face_predictions
+    assert len(face_predictions) == 1
+
+    face_prediction = face_predictions[0]
+    assert face_prediction.data["label"] == "face"
+    assert face_prediction.data["type"] == "face_annotation"
+    assert face_prediction.data["likelihood"] == 0.92
+    assert face_prediction.confidence == 0.92
+
+
+def test_flag_image_with_label_annotation_face():
+    ocr_json = {
+        "responses": [{"labelAnnotations": [{"description": "Face", "score": 0.8}]}]
+    }
+
+    ocr_result = OCRResult.from_json(ocr_json)
+    predictions = flag_image(ocr_result)
+
+    face_predictions = [
+        pred
+        for pred in predictions
+        if pred.type == PredictionType.image_flag
+        and pred.data.get("type") == "label_annotation"
+        and pred.data.get("label") == "face"
+    ]
+
+    assert face_predictions, "No face label predictions were detected"
+    assert len(face_predictions) == 1, "Expected exactly one face label prediction"
+    assert face_predictions[0].data["likelihood"] == 0.8
```

Comment:
You can directly check all values of the `Prediction`, doing a `assert face_predictions[0] == Prediction(...)`. In my opinion, it is good to perform a full check at least for one test case.